crypto/fipsmodule/evp/internal.h

```diff
@@ -77,14 +78,17 @@ extern "C" {
 // This is an implementation detail of |EVP_PKEY_HMAC|.
 #define EVP_MD_CTX_HMAC 0x0800
 
-typedef struct evp_pkey_asn1_method_st EVP_PKEY_ASN1_METHOD;
 typedef struct evp_pkey_method_st EVP_PKEY_METHOD;
 
 struct evp_pkey_asn1_method_st {
   int pkey_id;
   uint8_t oid[11];
   uint8_t oid_len;
 
+  int pkey_base_id;
```

Comment:
It seems like all values of `pkey_base_id` are identical to `pkey_id`. Can we omit this field and continue using `pkey_id` instead?

For a bit more context, `pkey_base_id` was part of pointing the historically different NID references to the same key types (`EVP_PKEY_RSA`/`EVP_PKEY_RSA2` and `EVP_PKEY_DSA1-4`). We don't have support for all the historical cruft, so I think we can get away with just the singular field.
* https://github.com/openssl/openssl/blob/8781087a69934cf45e201ac425d593b0e12a1538/crypto/rsa/rsa_ameth.c#L1009-L1010
* https://github.com/openssl/openssl/blob/8781087a69934cf45e201ac425d593b0e12a1538/crypto/dsa/dsa_ameth.c#L521-L541